components/pedigree/resources/src/main/resources/pedigree/view/datepicker.js

```diff
@@ -1,276 +1,445 @@
+  *
+  * @class AbstractFuzzyDateInput
+  * @constructor
+  * @param {element} A text input element. The behaviour of the element will be modified.
+  * @param {alwayEnabled} Booolean indicating whether this input should be enabled at all times
+  */
+  var AbstractFuzzyDateInput = Class.create({
+    initialize : function(element, alwaysEnabled){
+      this.element = element;
+      this.alwaysEnabled = alwaysEnabled;
+
+      //The isFuzzy boolean specifies whether the value contained in the iput is specific or fuzzy(a range, decade etc.). Blank and invalid values are not fuzzy.
+      this.isFuzzy = false;
+      this.isValid = true;
+
+      if (this.alwaysEnabled) {
+        this.enable();
+      } else {
+        this.disable();
+      }
+      this.element.onkeypress = this.onKeyPress.bind(this);
+      this.element.observe("focus", this.onFocus.bind(this));
+      this.element.observe("blur", this.onBlur.bind(this));
+      this.element.observe("keydown", this.onKeyDown.bind(this));
+      this.element.observe("keyup", this.onKeyUp.bind(this));
+    },
+    getValue : function() {
+      return this.element.value;
+    },
+    isEmpty : function() {
+      return this.getValue() === "";
+    },
+    setValue : function (value) {
+      value && this.enable();
+      var newVal = this.formatValue(value);
+      this.element.setValue(newVal);
+    },
+    getElement : function() {
+      return this.element;
+    },
+    onDateChange : function() {
+      this.refreshAvailableValues();
+      this.validate();
+      this.element.toggleClassName("invalid", (!this.isValid && this.getValue() !== ""));
+    },
+    enable: function() {
+      this.refreshAvailableValues();
+      this.element.enable();
+    },
+    disable: function() {
+      this.element.setValue("");
+      this.element.removeClassName("invalid");
+      this.isValid = true;
+      this.element.disable();
+    },
+    onFocus : function(e) {
+      this.element.removeClassName("invalid");
     },
+    onBlur : function(e) {
+      this.onValueSelected(e);
+    },
+    onValueSelected : function() {
+      this.element.fire("datepicker:date:changed");
+    },
+    onKeyDown : function(e) {
+      var charCode = e.key || e.which;
+      if (charCode == 9) {
+        this.onValueSelected(e);
+      }
+    },
+    onKeyUp: function(e) {
+      //if the input is valid on keyUp the input will consider the date selected before the user tabs away
+      this.validate() && this.onValueSelected();
+    },
+<<<<<<< HEAD
+
+=======
+>>>>>>> PT-2274: Date is now saved before blur
+    onKeyPress : function(e, regexp, maxLength, padZeros) {
+      var charCode = e.charCode || e.keyCode;
+      var character = String.fromCharCode(charCode);
+      //Call select handler on tab press.
+      if(charCode == 9){
+        this.onValueSelected();
+        return true;
+      }
+      //Allow non-character keys always
+      if (character === "") {return true;}
```

Comment:
It's not overkill. Many advocate for always using `===`, unless type coercion is explicitly wanted, since it helps avoid certain types of bugs, and has marginally better performance.